### ISSUE TITLE
[a11y] Cannot log in as Guest user with the "No Full Access" user role

### ISSUE THREAD

**Issue by grantfitzsimmons** [specify/specify7#1360]
**Describe the bug**
Cannot log in as "grant" guest user with the "No Full Access" user role.

**To Reproduce**
1. Go to freshfish
2. Try to log in as "grant"
3. See insufficient privileges

Potentially important, I changed the owner of a WB data set to "grant" before I tried to log in. This may trigger an issue since the user does not have access to the WB.

Was able to log in 2 days ago. Voucher collection.

**Screenshots**
![image](https://user-images.githubusercontent.com/37256050/165970370-cde1e973-8d8f-4d7a-87c3-b83472ff62e0.png)

**Desktop (please complete the following information):**
 - OS: macOS, Edge

**Comment by grantfitzsimmons** [comment]
![image](https://user-images.githubusercontent.com/37256050/165970989-498a4155-ae8c-43d4-ab90-7e9d69c9ad02.png)

Am able to access this in edge

**Comment by maxxxxxdlp** [comment]
The error dialog should show tell with permission you are lacking. It doesn't because of this bug: https://github.com/specify/specify7/issues/1386

You are probably lacking `create` permissions to the SpAppResourceDir, SpAppResource and SpAppResourceData and `modify` to SpAppResourceData tables. Those are not given by default to the non full-access users, but are required for user preferences to work.

Ben and I are working on a solution that wouldn't require giving those permissions, while still having user preferences work

**Comment by maxxxxxdlp** [comment]
Fixed in 519fe46b
